File: src/main/java/io/github/oleksivio/telegram/bot/api/model/method/message/DeleteMessage.java
package io.github.oleksivio.telegram.bot.api.model.method.message;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.oleksivio.telegram.bot.api.model.NetworkError;
import io.github.oleksivio.telegram.bot.core.controller.network.ActionNetworker;
import io.github.oleksivio.telegram.bot.core.model.method.ChatAction;
import io.github.oleksivio.telegram.bot.core.model.responses.BooleanResponse;
import io.github.oleksivio.telegram.bot.core.model.responses.CommonResponse;

/**
 * @see <a href="https://core.telegram.org/bots/api#deletemessage">deleteMessage</a>
 */
public class DeleteMessage extends ChatAction<Boolean> {
    private static final String METHOD = "deleteMessage";

    /**
     * message_id Integer Identifier of the message to delete
     */
    @JsonProperty("message_id")
    private Long messageId = null;

    public DeleteMessage(ActionNetworker actionNetworker) {
        super(METHOD, actionNetworker);
    }

    @Override
    protected Class<? extends CommonResponse<Boolean>> getResultWrapperClass() {
        return BooleanResponse.class;
    }

    public Long getMessageId() {
        return messageId;
    }

    public DeleteMessage setMessageId(Long messageId) {
        this.messageId = messageId;
        return this;
    }

    @Override
    public DeleteMessage setChatId(Long chatId) {
        pSetChatId(chatId);
        return this;
    }

    @Override
    public DeleteMessage setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/model/method/message/DeleteMessage.kt
package io.github.oleksivio.telegram.bot.api.model.method.message

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.telegram.bot.core.model.ApiDict
import io.github.oleksivio.telegram.bot.core.model.BooleanResponse
import io.github.oleksivio.telegram.bot.core.model.method.common.ChatAction

/**
 * @see [deleteMessage](https://core.telegram.org/bots/api/#deletemessage)
 */
data class DeleteMessage(
        /**
         * message_id Integer Yes Identifier of the message to delete
         */
        @JsonProperty(ApiDict.MESSAGE_ID_KEY)
        val messageId: Long,
        /**
         * chat_id Integer or String Yes Unique identifier for the target chat or username of the target channel (in the format @channelusername)
         */
        @JsonProperty(ApiDict.CHAT_ID_KEY)
        override val chatId: Long
) : ChatAction<Boolean>() {
    @JsonProperty(ApiDict.METHOD_KEY)
    override val method: String = "deleteMessage"

    @JsonIgnore
    override val resultWrapperClass = BooleanResponse::class
}
